error_test.go

```diff
@@ -45,10 +43,158 @@ func TestError(t *testing.T) {
 			if tc.err.Error() != tc.expected {
 				t.Errorf("expected err.Error() to be '%s' but was '%s'", tc.expected, tc.err.Error())
 			}
-			uuidErr := &ErrUUID{}
+			var uuidErr Error
 			if !errors.As(tc.err, &uuidErr) {
 				t.Error("expected errors.As() to work")
 			}
 		})
 	}
 }
+
+func TestAllErrorMessages(t *testing.T) {
+	tcs := []struct {
+		function string
+		uuidStr  string
+		expected string
+	}{
+		{ // 34 chars - With brackets
+			function: "parse",
+			uuidStr:  "..................................",
+			expected: "uuid: incorrect UUID format in string \"..................................\"",
+		},
+		{ // 41 chars - urn:uuid:
+			function: "parse",
+			uuidStr:  "123456789................................",
+			expected: "uuid: incorrect UUID format in string \"123456789\"",
+		},
+		{ // other
+			function: "parse",
+			uuidStr:  "....",
+			expected: "uuid: incorrect UUID length 4 in string \"....\"",
+		},
+		{ // 36 chars - canonical, but not correct format
+			function: "parse",
+			uuidStr:  "....................................",
+			expected: "uuid: incorrect UUID format in string \"....................................\"",
+		},
+		{ // 36 chars - canonical, invalid data
+			function: "parse",
+			uuidStr:  "xx00ae9e-dae3-459f-ad0e-6b574be3f950",
+			expected: "uuid: invalid UUID format",
+		},
+		{ // Hash like
+			function: "parse",
+			uuidStr:  "................................",
+			expected: "uuid: invalid UUID format",
+		},
+		{ // Hash like, invalid
+			function: "parse",
+			uuidStr:  "xx00ae9edae3459fad0e6b574be3f950",
+			expected: "uuid: invalid UUID format",
+		},
+		{ // Hash like, invalid
+			function: "parse",
+			uuidStr:  "xx00ae9edae3459fad0e6b574be3f950",
+			expected: "uuid: invalid UUID format",
+		},
+	}
+	for i, tc := range tcs {
+		t.Run(fmt.Sprintf("Test case %d", i), func(t *testing.T) {
+			id := UUID{}
+			err := id.Parse(tc.uuidStr)
+			if err == nil {
+				t.Error("expected an error")
+				return
+			}
+			if err.Error() != tc.expected {
+				t.Errorf("unexpected error '%s' != '%s'", err.Error(), tc.expected)
+			}
+			err = id.UnmarshalText([]byte(tc.uuidStr))
+			if err == nil {
+				t.Error("expected an error")
+				return
+			}
+			if err.Error() != tc.expected {
+				t.Errorf("unexpected error '%s' != '%s'", err.Error(), tc.expected)
+			}
+		})
+	}
+
+	// Unmarshal binary
+	id := UUID{}
+	b := make([]byte, 33)
+	expectedErr := "uuid: UUID must be exactly 16 bytes long, got 33 bytes"
+	err := id.UnmarshalBinary([]byte(b))
+	if err == nil {
+		t.Error("expected an error")
+		return
+	}
+	if err.Error() != expectedErr {
+		t.Errorf("unexpected error '%s' != '%s'", err.Error(), expectedErr)
+	}
+
+	// no hw address error
+	netInterfaces = func() ([]net.Interface, error) {
```

Comment:
This will potentially break parallel tests.  Would be better to move it to it's own test with a callout that it needs to modify the global state.